src/Query/Admin/VersionQuery/DbalVersionQuery.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Buddy\Repman\Query\Admin\VersionQuery;
+
+use Buddy\Repman\Query\Admin\VersionQuery;
+use Doctrine\DBAL\Connection;
+
+final class DbalVersionQuery implements VersionQuery
+{
+    private Connection $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function oldDistsCount(int $daysOld): int
+    {
+        return (int) $this
+            ->connection
+            ->fetchColumn(
+                'SELECT COUNT(v.id)
+                FROM organization_package_version v
+                JOIN organization_package p ON p.id = v.package_id
+                JOIN organization o ON o.id = p.organization_id
+                WHERE p.last_sync_at::date <= :date
```

Comment:
I think we can skip this condition, it will be better to delete versions that were simply not downloaded in the last X days